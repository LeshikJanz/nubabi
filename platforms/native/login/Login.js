// @flow
import React, { Component } from 'react';
import {
  Dimensions,
  Image,
  Platform,
  StyleSheet,
  TextInput,
  View,
} from 'react-native';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import { connect } from 'react-redux';
import theme, { NUBABI_RED } from '../../../core/themes/defaultTheme';
import { Link, SubmitButton, Text } from '../components';
import { loginRequest } from '../../../core/auth/actions';

const background = require('../../../core/images/loginBackground.jpg');
const logo = require('../../../core/images/logo.png');

const window = Dimensions.get('window');

type Props = {
  loginRequest: typeof loginRequest,
  isFetching: boolean,
};

export class Login extends Component {
  props: Props;

  state = {
    email: '',
    password: '',
  };

  passwordInput = null;
  submitButton = null;

  login = () => {
    const { email, password } = this.state;
    this.props.loginRequest(email, password);
  };

  loginViaKeyboard = () => {
    if (this.submitButton) {
      this.submitButton.touchableHandlePress();
    }
  };

  render() {
    const { isFetching } = this.props;

    return (
      <View style={styles.container}>
        <Image source={background} style={styles.background} />
        <View style={styles.backgroundFilter} />
        <KeyboardAwareScrollView
          style={styles.container}
          contentContainerStyle={styles.scrollContainer}
          keyboardShouldPersistTaps="handled"
        >
          <View style={{ flex: 1 }}>
            <Image source={logo} style={styles.logo} resizeMode="stretch" />

            <View style={styles.inputOuterContainer}>
              <View style={styles.inputContainer}>
                <TextInput
                  style={styles.textInput}
                  value={this.state.email}
                  placeholder="Email"
                  placeholderTextColor="rgba(255,255,255,.78)"
                  selectionColor="white"
                  keyBoardType="email-address"
                  autoCapitalize="none"
                  autoCorrect={false}
                  returnKeyType="next"
                  blurOnSubmit={false}
                  underlineColorAndroid="#eff1f7"
                  onChangeText={email => this.setState({ email })}
                  onSubmitEditing={() =>
                    this.passwordInput && this.passwordInput.focus()
                  }
                />
              </View>
              <View style={styles.inputContainer}>
                <TextInput
                  style={styles.textInput}
                  value={this.state.password}
                  ref={input => {
                    this.passwordInput = input;
                  }}
                  secureTextEntry
                  autoCapitalize="none"
                  placeholder="Password"
                  placeholderTextColor="rgba(255,255,255,.78)"
                  selectionColor="white"
                  autoCorrect={false}
                  onChangeText={password => this.setState({ password })}
                  returnKeyType="go"
                  onSubmitEditing={this.loginViaKeyboard}
                />
              </View>
              <SubmitButton
                innerRef={ref => (this.submitButton = ref)}
                animatedWidth={40}
                onPress={this.login}
                loading={isFetching}
                submitText="LOG IN"
                containerStyle={{
                  marginTop: 50,
                }}
                buttonStyle={{
                  width: 139,
                  height: 40,
                  borderRadius: 20,
                }}
                textStyle={{
                  fontSize: 17,
                  fontFamily: 'SF Pro Display',
                }}
              />
            </View>
          </View>
          <View style={{ alignItems: 'center', marginBottom: 15 }}>
            <Text size={1} color="white" style={() => ({ marginBottom: 5 })}>
              Don't have an account yet?
            </Text>

            <View style={{ flexDirection: 'row' }}>
              <Text color="white">Sign up for free on</Text>
              <Link
                title="Nubabi.com"
                color="white"
                url="https://nubabi.com"
                textStyle={{
                  marginLeft: 5,
                  fontWeight: theme.text.bold,
                }}
              />
            </View>
          </View>
        </KeyboardAwareScrollView>
      </View>
    );
  }
}

const mapDispatchToProps = {
  loginRequest,
};

const mapStateToProps = state => {
  return {
    isFetching: state.auth.isFetching || state.babies.isFetching,
    auth: state.auth,
  };
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'transparent',
  },
  scrollContainer: {
    flex: 1,
    justifyContent: 'center',
    backgroundColor: 'transparent',
  },
  errorContainer: {
    justifyContent: 'center',
    backgroundColor: theme.colors.primary,
    paddingTop: 20,
    paddingBottom: 10,
    position: 'absolute',
    top: 0,
    left: 0,
    flex: 1,
    width: window.width,
  },
  errorText: {
    color: theme.colors.white,
    textAlign: 'center',
  },
  background: {
    flex: 1,
    position: 'absolute',
    top: 0,
    width: window.width,
    height: window.height,
  },
  backgroundFilter: {
    flex: 1,
    position: 'absolute',
    top: 0,
    width: window.width,
    height: window.height,
    backgroundColor: 'rgba(116,130,148,0.55)',
  },
  logo: {
    position: 'absolute',
    alignItems: 'center',
    top: 140,
    width: 354 * 0.6,
    height: 112 * 0.6,
    marginLeft: (window.width - 354 * 0.6) / 2,
  },
  inputOuterContainer: {
    marginTop: 250,
  },
  inputContainer: {
    // flex: 1,
    flexDirection: 'column',
    marginBottom: 15,
    paddingBottom: 5,
    marginHorizontal: 30,
    borderColor: '#eff1f7',
    ...Platform.select({
      ios: {
        borderBottomWidth: 1,
        height: 40,
      },
    }),
  },
  inputLabel: {
    fontSize: 12,
    color: '#fff',
    marginBottom: 6,
  },
  textInput: {
    flex: 1,
    color: '#fff',
    fontSize: 18,
    lineHeight: 24,
    letterSpacing: -0.41,
  },
  submitButtonContainer: {
    marginTop: 20,
    flex: 1,
    alignItems: 'center',
    marginBottom: 50,
  },
  submitButton: {
    backgroundColor: NUBABI_RED,
    alignItems: 'center',
    justifyContent: 'center',
    width: 100,
    height: 30,
    borderRadius: 15,
  },
  submitText: {
    backgroundColor: 'transparent',
    fontSize: 14,
    fontWeight: '700',
    color: '#fff',
  },
  loader: {
    position: 'absolute',
    top: 30,
    right: 10,
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);
