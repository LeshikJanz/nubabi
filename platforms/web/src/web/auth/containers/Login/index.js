import React from 'react';
import reduxForm from 'redux-form/es/reduxForm';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { Title } from 'web/elements';
import { Helmet } from 'react-helmet';
import { parse } from 'qs';
import { Redirect } from 'react-router-dom';
import styled from 'styled-components';
import type { State, Dispatch } from 'types';
import LoginForm from 'web/auth/components/LoginForm';
import { loginRequest } from 'core/auth/actions';

const Wrapper = styled.div`
  padding: 10px;
`;

type LoginProps = {
  handleSubmit: () => void,
  pristine: boolean,
  reset: boolean,
  submitting: boolean,
  login: () => void,
  location: Object,
  isAuthenticated: boolean,
};

export const Login = (props: LoginProps) => {
  const {
    handleSubmit,
    pristine,
    reset,
    submitting,
    login,
    location,
    isAuthenticated,
  } = props;
  const handleFormSubmit = ({ username, password }) =>
    login(username, password);
  const { search } = location;
  let redirect;
  if (search.startsWith('?')) {
    const searchs = search.substring(1);
    const params = parse(searchs);
    if (params.redirect && params.redirect.length > 0)
      redirect = params.redirect;
  }
  const doRedirect = redirect !== undefined && isAuthenticated;
  return doRedirect ? (
    <Redirect to={redirect} />
  ) : (
    <Wrapper>
      <Helmet>
        <title>Nubabi | Login</title>
      </Helmet>
      <Title>Login</Title>
      <LoginForm
        handleSubmit={handleSubmit(handleFormSubmit)}
        pristine={pristine}
        submitting={submitting}
        reset={reset}
      />
    </Wrapper>
  );
};

const mapDispatchToProps = (dispatch: Dispatch) => ({
  login: (username, password) => {
    dispatch(loginRequest(username, password));
  },
});

const mapStateToProps = ({ auth }: State) => ({
  isAuthenticated: auth.isAuthenticated,
});

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(
    reduxForm({
      form: 'login',
    })(Login),
  ),
);
