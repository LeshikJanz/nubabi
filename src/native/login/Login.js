import React, { Component } from 'react';
import {
  View,
  Text,
  TextInput,
  StyleSheet,
  Image,
  Dimensions,
  TouchableHighlight,
  Platform,
} from 'react-native';
import {
  KeyboardAwareScrollView,
} from 'react-native-keyboard-aware-scroll-view';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import theme, { NUBABI_RED } from '../../common/themes/defaultTheme';
import * as loginActions from '../../common/auth/actions';
import Loader from '../components/Loader';

const background = require('../../common/images/loginBackground.jpg');
const logo = require('../../common/images/loginLogo.png');

const window = Dimensions.get('window');

type Props = {
  actions: any,
  isFetching: boolean,
};

export class Login extends Component {
  props: Props;

  state = {
    email: '',
    password: '',
  };

  login = () => {
    const { email, password } = this.state;
    this.props.actions.loginRequest(email, password);
  };

  renderLoader() {
    if (this.props.isFetching) {
      const type = Platform.select({
        ios: 'Arc',
        android: 'FadingCircle',
      });

      return (
        <Loader
          style={styles.loader}
          size={40}
          type={type}
          color={theme.colors.white}
        />
      );
    }

    return null;
  }

  render() {
    const { isFetching } = this.props;

    const buttonText = isFetching ? 'Logging in...' : 'LOG IN';
    return (
      <View style={styles.container}>
        <Image source={background} style={styles.background} />
        <View style={styles.backgroundFilter} />
        <KeyboardAwareScrollView
          style={styles.container}
          contentContainerStyle={styles.scrollContainer}
          keyboardShouldPersistTaps="handled"
        >
          <View style={{ flex: 1 }}>
            <Image source={logo} style={styles.logo} />

            <View style={styles.inputOuterContainer}>
              <View style={styles.inputContainer}>
                <Text style={styles.inputLabel}>EMAIL</Text>
                <TextInput
                  style={styles.textInput}
                  value={this.state.email}
                  placeholder="name@example.com"
                  keyBoardType="email-address"
                  autoFocus
                  autoCapitalize="none"
                  autoCorrect={false}
                  returnKeyType="next"
                  blurOnSubmit={false}
                  underlineColorAndroid="#eff1f7"
                  onChangeText={email => this.setState({ email })}
                  onSubmitEditing={() => this.passwordInput.focus()}
                />
              </View>
              <View style={styles.inputContainer}>
                <Text style={styles.inputLabel}>PASSWORD</Text>
                <TextInput
                  style={styles.textInput}
                  value={this.state.password}
                  ref={input => {
                    this.passwordInput = input;
                  }}
                  secureTextEntry
                  autoCapitalize="none"
                  palceholder="password"
                  autoCorrect={false}
                  focus={this.state.focusPassword}
                  onChangeText={password => this.setState({ password })}
                  returnKeyType="go"
                  onSubmitEditing={this.login}
                />
              </View>
              <TouchableHighlight
                underlayColor="rgba(0,0,0,0)"
                style={styles.oneButton}
                onPress={this.login}
                disabled={isFetching}
              >
                <View style={styles.submitButtonContainer}>
                  <View style={styles.submitButton}>
                    <Text style={styles.submitText}>{buttonText}</Text>
                  </View>
                </View>
              </TouchableHighlight>
            </View>
            {this.renderLoader()}

          </View>
        </KeyboardAwareScrollView>
      </View>
    );
  }
}

const mapDispatchToProps = dispatch => {
  return {
    actions: bindActionCreators(loginActions, dispatch),
  };
};

const mapStateToProps = state => {
  return {
    isFetching: state.auth.isFetching || state.babies.isFetching,
    auth: state.auth,
  };
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'transparent',
  },
  scrollContainer: {
    flex: 1,
    justifyContent: 'center',
    backgroundColor: 'transparent',
  },
  errorContainer: {
    justifyContent: 'center',
    backgroundColor: theme.colors.primary,
    paddingTop: 20,
    paddingBottom: 10,
    position: 'absolute',
    top: 0,
    left: 0,
    flex: 1,
    width: window.width,
  },
  errorText: {
    color: theme.colors.white,
    textAlign: 'center',
  },
  background: {
    flex: 1,
    position: 'absolute',
    top: 0,
    width: window.width,
    resizeMode: 'cover',
    height: window.height,
  },
  backgroundFilter: {
    flex: 1,
    position: 'absolute',
    top: 0,
    width: window.width,
    height: window.height,
    backgroundColor: 'rgba(116,130,148,0.55)',
  },
  logo: {
    alignItems: 'center',
    marginTop: 100,
    width: 370 * 0.6,
    height: 122 * 0.6,
    marginLeft: (window.width - 370 * 0.6) / 2,
    resizeMode: 'stretch',
  },
  inputOuterContainer: {
    marginTop: 150,
    //flex: 1,
  },
  inputContainer: {
    // flex: 1,
    flexDirection: 'column',
    marginBottom: 15,
    paddingBottom: 5,
    marginHorizontal: 30,
    borderColor: '#eff1f7',
    ...Platform.select({
      ios: {
        borderBottomWidth: 1,
        height: 50,
      },
    }),
  },
  inputLabel: {
    fontSize: 12,
    color: '#fff',
    marginBottom: 6,
  },
  textInput: {
    flex: 1,
    color: '#fff',
    fontSize: 18,
    lineHeight: 26,
  },
  submitButtonContainer: {
    marginTop: 20,
    flex: 1,
    alignItems: 'center',
    marginBottom: 50,
  },
  submitButton: {
    backgroundColor: NUBABI_RED,
    alignItems: 'center',
    justifyContent: 'center',
    width: 100,
    height: 30,
    borderRadius: 15,
  },
  submitText: {
    backgroundColor: 'transparent',
    fontSize: 14,
    fontWeight: '700',
    color: '#fff',
  },
  loader: {
    position: 'absolute',
    top: 30,
    right: 10,
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);
