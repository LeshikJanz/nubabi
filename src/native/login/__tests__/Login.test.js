import 'react-native';
import React from 'react';
import { Login } from '../Login';
import renderer from 'react-test-renderer';

jest.mock('../../components/Alert');

test('renders correctly', () => {
  const tree = renderer
    .create(<Login isFetching={false} actions={{}} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('show progress when logging in', () => {
  const tree = renderer.create(<Login isFetching actions={{}} />).toJSON();

  expect(tree).toMatchSnapshot();
});
