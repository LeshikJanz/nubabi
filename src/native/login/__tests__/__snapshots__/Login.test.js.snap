// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
style = Object {
    "borderColor": "#9EABBC",
    "borderBottomWidth": 0,
    "borderLeftWidth": 0,
    "borderRightWidth": 0,
    "borderTopWidth": 0,
    "fontSize": 13,
    "lineHeight": 16,
    "color": "#FFFFFF",
    "fontFamily": "SF Pro Text",
    "marginBottom": 5,
    "marginLeft": 5,
    "fontWeight": "700"
}

<View
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
>
  <Image
    resizeMode="cover"
    source={1}
    style={
      Object {
        "flex": 1,
        "height": 1334,
        "position": "absolute",
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "rgba(116,130,148,0.55)",
        "flex": 1,
        "height": 1334,
        "position": "absolute",
        "top": 0,
        "width": 750,
      }
    }
  />
  <RCTScrollView
    contentContainerStyle={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    onScroll={[Function]}
    scrollEventThrottle={0}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Image
          resizeMode="stretch"
          source={1}
          style={
            Object {
              "alignItems": "center",
              "height": 73.2,
              "marginLeft": 264,
              "marginTop": 135,
              "width": 222,
            }
          }
        />
        <View
          style={
            Object {
              "marginTop": 150,
            }
          }
        >
          <View
            style={
              Object {
                "borderBottomWidth": 1,
                "borderColor": "#eff1f7",
                "flexDirection": "column",
                "height": 40,
                "marginBottom": 15,
                "marginHorizontal": 30,
                "paddingBottom": 5,
              }
            }
          >
            <TextInput
              autoCapitalize="none"
              autoCorrect={false}
              blurOnSubmit={false}
              keyBoardType="email-address"
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Email"
              placeholderTextColor="rgba(255,255,255,.78)"
              returnKeyType="next"
              selectionColor="white"
              style={
                Object {
                  "color": "#fff",
                  "flex": 1,
                  "fontSize": 18,
                  "letterSpacing": -0.41,
                  "lineHeight": 24,
                }
              }
              underlineColorAndroid="#eff1f7"
              value=""
            />
          </View>
          <View
            style={
              Object {
                "borderBottomWidth": 1,
                "borderColor": "#eff1f7",
                "flexDirection": "column",
                "height": 40,
                "marginBottom": 15,
                "marginHorizontal": 30,
                "paddingBottom": 5,
              }
            }
          >
            <TextInput
              autoCapitalize="none"
              autoCorrect={false}
              focus={undefined}
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Password"
              placeholderTextColor="rgba(255,255,255,.78)"
              returnKeyType="go"
              secureTextEntry={true}
              selectionColor="white"
              style={
                Object {
                  "color": "#fff",
                  "flex": 1,
                  "fontSize": 18,
                  "letterSpacing": -0.41,
                  "lineHeight": 24,
                }
              }
              value=""
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                  "marginTop": 20,
                },
                Object {},
                Object {
                  "marginTop": 50,
                },
              ]
            }
          >
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#EA3154",
                  "borderRadius": 20,
                  "height": 40,
                  "justifyContent": "center",
                  "width": 139,
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#fff",
                        "fontSize": 12,
                        "fontWeight": "700",
                      },
                      Object {
                        "fontFamily": "SF Pro Display",
                        "fontSize": 17,
                      },
                    ]
                  }
                >
                  LOG IN
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginBottom": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "borderBottomWidth": 0,
              "borderColor": "#9EABBC",
              "borderLeftWidth": 0,
              "borderRightWidth": 0,
              "borderTopWidth": 0,
              "color": "#FFFFFF",
              "fontFamily": "SF Pro Text",
              "fontSize": 13.866666666666667,
              "lineHeight": 16,
              "marginBottom": 5,
            }
          }
        >
          Don't have an account yet?
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "borderBottomWidth": 0,
                "borderColor": "#9EABBC",
                "borderLeftWidth": 0,
                "borderRightWidth": 0,
                "borderTopWidth": 0,
                "color": "#FFFFFF",
                "fontFamily": "SF Pro Text",
                "fontSize": 13,
                "lineHeight": 16,
              }
            }
          >
            Sign up for free on
          </Text>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "borderBottomWidth": 0,
                  "borderColor": "#9EABBC",
                  "borderLeftWidth": 0,
                  "borderRightWidth": 0,
                  "borderTopWidth": 0,
                  "color": "#FFFFFF",
                  "fontFamily": "SF Pro Text",
                  "fontSize": 13,
                  "fontWeight": "700",
                  "lineHeight": 16,
                  "marginLeft": 5,
                }
              }
            >
              Nubabi.com
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`show progress when logging in 1`] = `
style = Object {
    "borderColor": "#9EABBC",
    "borderBottomWidth": 0,
    "borderLeftWidth": 0,
    "borderRightWidth": 0,
    "borderTopWidth": 0,
    "fontSize": 13,
    "lineHeight": 16,
    "color": "#FFFFFF",
    "fontFamily": "SF Pro Text",
    "marginBottom": 5,
    "marginLeft": 5,
    "fontWeight": "700"
}

<View
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
>
  <Image
    resizeMode="cover"
    source={1}
    style={
      Object {
        "flex": 1,
        "height": 1334,
        "position": "absolute",
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "rgba(116,130,148,0.55)",
        "flex": 1,
        "height": 1334,
        "position": "absolute",
        "top": 0,
        "width": 750,
      }
    }
  />
  <RCTScrollView
    contentContainerStyle={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    onScroll={[Function]}
    scrollEventThrottle={0}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Image
          resizeMode="stretch"
          source={1}
          style={
            Object {
              "alignItems": "center",
              "height": 73.2,
              "marginLeft": 264,
              "marginTop": 135,
              "width": 222,
            }
          }
        />
        <View
          style={
            Object {
              "marginTop": 150,
            }
          }
        >
          <View
            style={
              Object {
                "borderBottomWidth": 1,
                "borderColor": "#eff1f7",
                "flexDirection": "column",
                "height": 40,
                "marginBottom": 15,
                "marginHorizontal": 30,
                "paddingBottom": 5,
              }
            }
          >
            <TextInput
              autoCapitalize="none"
              autoCorrect={false}
              blurOnSubmit={false}
              keyBoardType="email-address"
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Email"
              placeholderTextColor="rgba(255,255,255,.78)"
              returnKeyType="next"
              selectionColor="white"
              style={
                Object {
                  "color": "#fff",
                  "flex": 1,
                  "fontSize": 18,
                  "letterSpacing": -0.41,
                  "lineHeight": 24,
                }
              }
              underlineColorAndroid="#eff1f7"
              value=""
            />
          </View>
          <View
            style={
              Object {
                "borderBottomWidth": 1,
                "borderColor": "#eff1f7",
                "flexDirection": "column",
                "height": 40,
                "marginBottom": 15,
                "marginHorizontal": 30,
                "paddingBottom": 5,
              }
            }
          >
            <TextInput
              autoCapitalize="none"
              autoCorrect={false}
              focus={undefined}
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Password"
              placeholderTextColor="rgba(255,255,255,.78)"
              returnKeyType="go"
              secureTextEntry={true}
              selectionColor="white"
              style={
                Object {
                  "color": "#fff",
                  "flex": 1,
                  "fontSize": 18,
                  "letterSpacing": -0.41,
                  "lineHeight": 24,
                }
              }
              value=""
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                  "marginTop": 20,
                },
                Object {
                  "opacity": 0.2,
                },
                Object {
                  "marginTop": 50,
                },
              ]
            }
          >
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#EA3154",
                  "borderRadius": 20,
                  "height": 40,
                  "justifyContent": "center",
                  "width": 139,
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginBottom": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "borderBottomWidth": 0,
              "borderColor": "#9EABBC",
              "borderLeftWidth": 0,
              "borderRightWidth": 0,
              "borderTopWidth": 0,
              "color": "#FFFFFF",
              "fontFamily": "SF Pro Text",
              "fontSize": 13.866666666666667,
              "lineHeight": 16,
              "marginBottom": 5,
            }
          }
        >
          Don't have an account yet?
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "borderBottomWidth": 0,
                "borderColor": "#9EABBC",
                "borderLeftWidth": 0,
                "borderRightWidth": 0,
                "borderTopWidth": 0,
                "color": "#FFFFFF",
                "fontFamily": "SF Pro Text",
                "fontSize": 13,
                "lineHeight": 16,
              }
            }
          >
            Sign up for free on
          </Text>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "borderBottomWidth": 0,
                  "borderColor": "#9EABBC",
                  "borderLeftWidth": 0,
                  "borderRightWidth": 0,
                  "borderTopWidth": 0,
                  "color": "#FFFFFF",
                  "fontFamily": "SF Pro Text",
                  "fontSize": 13,
                  "fontWeight": "700",
                  "lineHeight": 16,
                  "marginLeft": 5,
                }
              }
            >
              Nubabi.com
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
